examples/with-nextjs-and-clerk-auth/src/lib/monite-api/demo-data-generator/general.service.ts

```diff
@@ -31,16 +33,25 @@ export abstract class GeneralService {
   protected readonly entityId: string;
   protected readonly logger?: ILogger;
   protected readonly request: ReturnType<typeof createMoniteClient>;
+  private cachedEntity:
+    | components['schemas']['EntityOrganizationResponse']
+    | undefined;
 
   constructor(params: IGeneralServiceConstructor) {
     this.token = params.token;
     this.entityId = params.entityId;
     this.logger = params.logger;
-    this.request = createMoniteClient({
-      headers: {
-        Authorization: `${params.token.token_type} ${params.token.access_token}`,
-      },
-    });
+    this.request = createMoniteClient(params.token);
+  }
+
+  protected async getEntity(): Promise<
+    components['schemas']['EntityOrganizationResponse']
+  > {
+    if (this.cachedEntity) return this.cachedEntity;
+
+    const entity = await this.request.getEntity(this.entityId);
```

Comment:
Please, don't implement any caching for the data generation process. it's not a production code, and potential issues with the cache invalidation will add more complexity than it's worth. One more `GET` is not a shame.